src/app/shell/admin-tools/data/history-log/history-log.component.ts

```diff
@@ -80,33 +73,49 @@ export class HistoryLogComponent implements OnInit, OnDestroy {
       .subscribe((searchString: string) => {
         if (this.searchFormControl.dirty) {
           this.searchString = searchString;
-          this.dispatchProperValue(this.tabIndex, this.filters, searchString);
+
+          this.currentPage = PaginationConstants.firstPage;
+          this.getTableData(searchString);
         }
       });
   }
 
   onTabChange(event: MatTabChangeEvent): void {
+    this.currentPage = PaginationConstants.firstPage;
     this.tabIndex = event.index;
-    this.dispatchProperValue(event.index);
+    this.getTableData();
 
     this.router.navigate(['./'], {
       relativeTo: this.route,
-      queryParams: { tab: HistoryLogTabsUkrReverse[event.tab.textLabel] },
+      queryParams: { tab: HistoryLogTabsUkrReverse[event.tab.textLabel] }
     });
   }
 
   onItemsPerPageChange(itemsPerPage: number): void {
-    this.store.dispatch([new SetTableItemsPerPage(itemsPerPage)]);
-    this.dispatchProperValue(this.tabIndex);
+    this.filters.size = itemsPerPage;
+    this.getTableData();
   }
 
   onPageChange(page: PaginationElement): void {
     this.currentPage = page;
-    this.store.dispatch([new OnPageChangeHistoryLog(page)]);
-    this.dispatchProperValue(this.tabIndex, this.filters, this.searchString);
+    this.getTableData(this.searchString);
   }
 
-  private dispatchProperValue(tabIndex: number, filters?: FilterData, searchString?: string): void {
+  onFilter(event: FilterData): void {
+    event.from = this.filters.from;
+    event.size = this.filters.size;
+    this.filters = event;
```

Comment:
there is definetely somethig wrong.
you assign to object1 fields from object 2, and then assign to object2 the object 1.
I see 105, and 106 should be removed.